src/Tools/babylon.assetsManager.ts

```diff
@@ -98,11 +82,14 @@ module BABYLON {
         }
     }
 
-    export class MeshAssetTask extends AbstractAssetTask implements IAssetTask<MeshAssetTask> {
+    export class MeshAssetTask extends AbstractAssetTask {

```

Comment:
What about this?
I guess this fix our issues :)

Reply:
pinging @RaananW

Reply:
Will this return MeshAssetTask  in the onSuccess callback? I will pull this soon locally and test, just wondering if you tested already.

Reply:
I did but would like to get a second pass

Reply:
Pulling now :) I just noticed you added onSuccess and onError to all tasks, this is probably a must.

Reply:
So! Works for me. The only thing we need to consider is that some people will have the interface referenced in their code (because of the way it used to be implemented), so it is a (very small! 😄 ) backwards compatibility issue.

Reply:
Agree. Not a big deal
I will add a note in the what's new
Lt me fix the <any> problem for observable and I'll do the final PR